### PR TITLE
Docker preperation and changes

### PR DESCRIPTION
**Pull request by FlipperLP**

### CODE REVIEW

**Review by Coder-Tavi — DISMISSED** [review]
Is it possible to retain the current .gitignore?

File: config.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (Coder-Tavi) [inline config.json:1]: Why is the config.json exposed here?

**Comment by FlipperLP** [comment]
it is possible, but .env needs to be added and config.json removed.

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (Coder-Tavi) [inline .vscode/launch.json:1]: Can we not expose this?

File: config.json (hunk 1 above)
Comment (FlipperLP) [reply config.json:1]: I let you know an hour ago, with no reply. Tokens and creds have been moved to .env

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# get node
+FROM node:current-buster-slim
```
Comment (Coder-Tavi) [inline Dockerfile:2]: Isn't `node alpine` something that we can use here?

File: package.json [hunk 4]
```diff
@@ -12,5 +12,23 @@
   },
   "devDependencies": {
     "eslint": "8.23.1"
-  }
+  },
```
Comment (Coder-Tavi) [inline package.json:15]: Gotta love how node forces you to add these details

File: .vscode/launch.json (hunk 2 above)
Comment (FlipperLP) [reply .vscode/launch.json:1]: it has the config to debug the bot. Unless you find out yourself on how to do it/explain it in the docs, the file can be removed again.

File: Dockerfile (hunk 3 above)
Comment (FlipperLP) [reply Dockerfile:2]: yes we can. i stuck to buster as it was working more stable for me. but i can change it

File: package.json (hunk 4 above)
Comment (FlipperLP) [reply package.json:15]: its useful information.

File: .dockerignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 node_modules
 npm-debug.log
 docker-compose.yml
-dist.*
\ No newline at end of file
+dist.*
+.env
```
Comment (Coder-Tavi) [inline .dockerignore:5]: *wheeeeeze* imagine leaving an env exposed

Comment (Coder-Tavi) [inline .dockerignore:5]: *wheeeeeze* imagine leaving an env exposed

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,3 @@
-/node_modules
-/.vscode
-config.json
\ No newline at end of file
+.vsls.json
```
Comment (Coder-Tavi) [inline .gitignore:1]: What's this?

File: .vscode/launch.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (Coder-Tavi) [inline .vscode/launch.json:4]: Remove the comments, if possible to keep it clean

File: config.json (hunk 1 above)
Comment (Coder-Tavi) [reply config.json:1]: Why are they present here, if I can ask then?

**Review by Coder-Tavi — CHANGES_REQUESTED** [review]
Last comments before push

File: .dockerignore (hunk 5 above)
Comment (Coder-Tavi) [reply .dockerignore:5]: (this was a joke comment)

File: .gitignore (hunk 6 above)
Comment (FlipperLP) [reply .gitignore:1]: i think that was already present from your end. but not sure

Comment (Coder-Tavi) [reply .gitignore]: Dunno what that is but let's just keep it because why not?

**Review by Coder-Tavi — APPROVED** [review]

**Pull request merged**